website/docs/d/policy_groups.html.markdown

```diff
@@ -0,0 +1,69 @@
+---
+subcategory: "Firewall"
+layout: "nsxt"
+page_title: "NSXT: policy_groups"
+description: A policy groups data source. This data source builds a list representation of the whole groups table, containing `id`, `display_name` and `path` attributes.
+---
+
+# nsxt_policy_groups
+
+This data source builds a list of the whole policy Groups table. Such list can be referenced in configuration to obtain object identifier attributes lookup by display name at a cost of single roundtrip to NSX, which improves apply and refresh
+time at scale, compared to multiple instances of `nsxt_policy_group` data source.
+
+## Example Usage
+
+```hcl
+data "nsxt_policy_groups" "grouplist" {
+}
+
+resource "nsxt_policy_predefined_security_policy" "test" {
+  path = data.nsxt_policy_security_policy.default_l3.path
+
+  tag {
+    scope = "color"
+    tag   = "orange"
+  }
+
+  rule {
+    display_name = "allow_icmp"
+    destination_groups = [
+      data.nsxt_policy_groups.test.items[index(data.nsxt_policy_groups.test.items.*.display_name, "Cats")].path,
```

Comment:
I feel this syntax might be too complex to be widely adopted.. perhaps it would be more useful to offer just display_name=>path map.